### ISSUE TITLE
AttributeError: 'function' object has no attribute 'output' on @protected routes

### ISSUE THREAD

**Issue by Berkodev** [ahopkins/sanic-jwt#89]
Hello,
ever since upgrading sanic_jwt to the most recent version (1.0.2.1 through pypi) I've been getting a strange bug when attempting to access @protected routes (from the sanic log):
> [2018-04-02 17:11:40 +0300] [3156] [ERROR] Invalid response object for url b'/test', Expected Type: HTTPResponse, Actual Type: <class 'function'>
[2018-04-02 17:11:40 +0300] [3156] [ERROR] Traceback (most recent call last):
  File "C:\Users\User1\AppData\Local\Programs\Python\Python36-32\lib\site-packages\sanic\server.py", line 342, in write_response
    response.output(
AttributeError: 'function' object has no attribute 'output'

I'm trying to access this by making a simple GET request to the URL, with a valid authentication header containing a previously generated JWT access token.

This is the code I'm using for the server-side (vastly simplified for the sake of reproducability):
```
import sanic_jwt
from sanic import Sanic, response
from sanic_jwt import protected

app = Sanic()


@app.route("/test")
@protected
async def test(request):
    return response.json('hello!')

if __name__ == "__main__":
    sanic_jwt.Initialize(app, authenticate=lambda x: True)
    app.run(host="127.0.0.1", port=8000, debug=True)
```
Note that if I comment out the @protected decorator, everything works fine.

I'm using ver. 0.7.0 on Python 3.6 and getting the same result both on Windows and Linux (different machines).
Will be happy to provide any other info that's missing.

Thanks in advance!

**Comment by vltr** [comment]
Thanks, @Berkodev !

I was able to reproduce this error here. Thanks for reporting and providing the smallest possible code to reproduce this. We'll look into it :wink:

**Comment by Berkodev** [comment]
I could have sworn I've tried adding the parenthesis earlier to non avail, but it indeed fixes the issue.

Thanks for the quick response! May have a go at trying to fix it later.

**Comment by vltr** [comment]
@Berkodev no problem! I'm glad it was just this :wink: If you have any problems down the road regarding this, please, let use know

**Comment by ahopkins** [comment]
@Berkodev Thanks for sharing your issue. This is something that I've had on my mind to fix and indeed will be something fixed for v1.1 on #68. Also, this is seen when using the decorator on a class based view.

**Comment by ahopkins** [comment]
@vltr To your point, I think the debate is simple. It should work either way, and will not take much for us to tweak it.
